### PR TITLE
Support newer versions of LLVM and ROCm

### PR DESCRIPTION
**Pull request by tbennun**
Adds support for newer versions of LLVM (18) and ROCm (6.2).

### CODE REVIEW

File: lib/JitEngine.cpp [hunk 1]
```diff
@@ -42,7 +43,11 @@ using namespace llvm;
 Expected<std::unique_ptr<TargetMachine>>
 JitEngine::createTargetMachine(Module &M, StringRef Arch, unsigned OptLevel) {
   Triple TT(M.getTargetTriple());
+#if LLVM_VERSION_MAJOR >= 18
+  std::optional<llvm::CodeGenOptLevel> CGOptLevel = CodeGenOpt::getLevel(OptLevel);
```
Comment (tbennun) [inline lib/JitEngine.cpp:47]: One could also just use `auto` here

Comment (ggeorgakoudis) [reply lib/JitEngine.cpp:47]: Let's go for `auto` to avoid preprocessor pollution

Comment (tbennun) [reply lib/JitEngine.cpp]: Done.

**Comment by tbennun** [comment]
> We need to have tests ran for LLVM 18 too. However, github and tioga tests succeeded on LLVM 17, which makes me believe that your changes are portable for both LLVM 17 and 18?

The functions I'm using (`PointerType::get`) are compatible going back to LLVM 14 (and perhaps even earlier).

I also addressed your comment and it still works for me.

**Comment by ggeorgakoudis** [comment]
Supereded #44

**Pull request closed without merging**